tensorflow/core/kernels/scatter_functor.h

```diff
@@ -205,17 +206,30 @@ struct ScatterFunctorBase {
     // indices and params sizes were validated in DoCompute().
     const Index N = static_cast<Index>(indices.size());
     const Index limit = static_cast<Index>(params.dimension(0));
-    for (Index i = 0; i < N; i++) {
-      // Grab the index and check its validity.  Do this carefully,
-      // to avoid checking the value and grabbing it again from
-      // memory a second time (a security risk since it may change in between).
-      const Index index = ::tensorflow::internal::SubtleMustCopy(indices(i));
-      if (!FastBoundsCheck(index, limit)) return i;
-      // Copy last Ndim-1 dimensions of updates[i] to params[index]
-      scatter_op::internal::Assign<op>::Run(params.template chip<0>(index),
-                                            updates.template chip<0>(i));
-    }
-    return -1;
+    mutex mu_;
+    Index bad_index = -1 GUARDED_BY(mu_);
```

Comment:
The assignment should be after `GUARDED_BY`.
```suggestion
    Index bad_index GUARDED_BY(mu_) = -1;
```